src/types/data_buffer.hpp

```diff
@@ -40,6 +40,7 @@ struct DataBuffer {
   }
 
   DataBuffer() : size(0) { value = new std::byte[DefaultBufferSize]; }
+  ~DataBuffer() { if(value != NULL) delete value; }
```

Comment:
GitHub Actions CI [still reports](https://github.com/nikezono/LineairDB/runs/8001431605?check_suite_focus=true) there exists alloc-dealloc-mismatch as the following:

```
4: ==4216==ERROR: AddressSanitizer: alloc-dealloc-mismatch (operator new [] vs operator delete) on 0x615000011600
```

At the above line L42 we allocate `value` via the operator `new []`.
To suppress this warning, use `delete[]` instead of `delete`.

Reply:
In addition, I recommend you to use `nullptr` instead of `NULL` for comparing pointers.
FYI: https://cpprefjp.github.io/lang/cpp11/nullptr.html